PCLConfig.cmake.in

```diff
@@ -117,6 +117,9 @@ macro(find_eigen3)
     set(Eigen3_DIR "${PCL_ROOT}/3rdParty/Eigen3/share/eigen3/cmake/")
   endif()
   find_package(Eigen3 3.3 REQUIRED NO_MODULE)
+  if(NOT EIGEN3_FOUND AND Eigen3_FOUND)
+    set(EIGEN3_FOUND ${Eigen3_FOUND})
+  endif()
```

Comment:
Is this also necessary here: https://github.com/PointCloudLibrary/pcl/blob/master/PCLConfig.cmake.in#L119 ?

Reply:
This is to make PCL configuration work when the user installs Eigen from source. PCLConfig checks eigen via `EIGEN3_FOUND` but `Eigen3_FOUND` is defined instead of `EIGEN3_FOUND` in the latest Eigen version 3.4.90.

Reply:
Thanks for the explanation! I was just wondering if it is necessary to also add these three lines after `find_package(Eigen3 3.3 REQUIRED NO_MODULE)` in `PCLConfig.cmake.in`. That file is used when CMake tries to find PCL from a user project. In line 315, it checks `if(${LIB}_FOUND)`, which would translate to `if(EIGEN3_FOUND)`, so I would assume that the problem you described exists here as well. Did you perhaps check whether a user project, for example one of the tutorials, can call `find_package(PCL REQUIRED)` successfully if Eigen 3.4.90 is installed?

Reply:
```
cmake_minimum_required(VERSION 3.20)
project(pcl_examples)
find_package(Eigen3 3.4.90 REQUIRED)
find_package(PCL 1.13 REQUIRED)
```

I test it with the above example and get the following error if I don't add those three lines to copy `Eigen3_FOUND` to `EIGEN3_FOUND`. 
```
[cmake] -- The C compiler identification is GNU 13.2.1
[cmake] -- The CXX compiler identification is GNU 13.2.1
[cmake] -- Detecting C compiler ABI info
[cmake] -- Detecting C compiler ABI info - done
[cmake] -- Check for working C compiler: /usr/bin/gcc - skipped
[cmake] -- Detecting C compile features
[cmake] -- Detecting C compile features - done
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /usr/bin/g++ - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] CMake Error at /usr/share/pcl-1.13/PCLConfig.cmake:75 (message):
[cmake]   common is required but eigen3 was not found
[cmake] Call Stack (most recent call first):
[cmake]   /usr/share/pcl-1.13/PCLConfig.cmake:386 (pcl_report_not_found)
[cmake]   /usr/share/pcl-1.13/PCLConfig.cmake:590 (find_external_library)
[cmake]   CMakeLists.txt:5 (find_package)
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
```

In the Eigen3Config.cmake of `3.4.0`, it says that `EIGEN3_FOUND` and other variables will be removed in the future. And in `3.4.90`, these variables are totally removed.

Reply:
Oops, sorry, I misinterpreted in which file you added this. You actually added it in PCLConfig.cmake.in.
In that case I think it is necessary to add the same three lines after https://github.com/PointCloudLibrary/pcl/blob/master/CMakeLists.txt#L322
I just tested with Eigen 3.4.90, and before adding these lines in PCL's CMakeLists.txt, CMake reports that it can't build the `common` module because eigen3 is not found, when I try to build PCL from source.

Reply:
To make `PCL_SUBSYS_DEPEND` handle `Eigen3` correctly, I think we should add these three lines to the top level CMakeLists.txt. I already push the changes. 

But when I test it with Eigen 3.4.90 and `cmake ..` before I add the three lines, CMake does not report error about building `common`. Perhaps I should post some options but I have no idea. The build also works without adding the three lines because [https://github.com/PointCloudLibrary/pcl/blob/master/common/CMakeLists.txt#182](https://github.com/PointCloudLibrary/pcl/blob/master/common/CMakeLists.txt#182) links `Eigen3::Eigen` to the `common` module.

Reply:
Okay, I know what happens. I test the build with `vtk` installed. Finding `vtk` will set `EIGEN3_FOUND` for us. But if we don't build with `vtk`, we do need those three lines to set `EIGEN3_FOUND` correctly.